mat/svd.go

```diff
@@ -20,6 +20,35 @@ type SVD struct {
 	vt blas64.General
 }
 
+// SVDKind specifies the treatment of singular vectors during an SVD
+// factorization.
+type SVDKind int
+
+const (
+	// SVDNone specifies that no singular vectors should be computed during
+	// the decomposition.
+	SVDNone SVDKind = 0
+
+	// SVDThinU specifies the thin decomposition of U should be computed.
+	SVDThinU SVDKind = 1 << (iota - 1)
+	// SVDThinU specifies the full decomposition of U should be computed.
```

Comment:
SVDFullU

Also strictly speaking it doesn't specify the _decomposition_ of U or V but ok.

Reply:
I changed "of" to "for". If you have a compact description that is strictly speaking correct I'm happy to change it... "SVDThinU specifies that when the input matrix is factorized, only the vectors of U associated with the non-rank deficient dimensions of the input matrix will be computed and stored". It's correct but it's about twice as long as probably twice as confusing.